src/plugins/misc/commands/jumbo.command.ts

```diff
@@ -0,0 +1,30 @@
+import {ChatInputCommandInteraction, inlineCode, PermissionFlagsBits} from "discord.js";
+import {SlashCommandBuilder, SlashCommandScope} from "../../../builders/SlashCommandBuilder";
+import {useChatCommand} from "../../../hooks/useChatCommand";
+
+const builder = new SlashCommandBuilder()
+    .setName("jumbo")
+    .setDescription("Make G'day blow up an image.")
+    .addStringOption(option => option.setName("emoji").setDescription("Emoji to jumbo size.").setRequired(true))
+    .setScope(SlashCommandScope.MAIN_GUILD);
+
+useChatCommand(builder, async (interaction: ChatInputCommandInteraction) => {
+    const emoji = interaction.options.getString('emoji', true).trim();
+    let emojiURL = null
+
+    if (emoji.startsWith("<") && emoji.endsWith(">")) {
+        const id = emoji.match(/\d{15,}/g);
+        if (!id) {
+            return "Invalid emoji format!";
+        }
+        const type = emoji.startsWith("<a:") ? "gif" : "png";
+
+        emojiURL = `https://cdn.discordapp.com/emojis/${id}.${type}?quality=lossless`;
+    };
+    
+    if (!emoji.startsWith("<" || "<a:")) {
```

Comment:
Cannot believe I missed this, but it does not work like that.

`||` is a boolean (works on true/false) operator that represents logical OR. That is `x || y` will be `true` if either x or y are `true`.  This operator is most commonly used in if statements, e.g. `if (x || y)` which will of course run everything in the statement if either x or y is true.

For optimisation, JavaScript uses something called short-circuit evaluation, meaning if `x` evaluates to true, it doesn't bother evaluating `y` because it knows `x || y` will be true no matter what y is. The interesting thing about that is we can use this (at least in JS) beyond a true/false value, which is what we're doing (unknowingly) here. The reason why is that everything can be used in place of a boolean™️ and will represent the truthy/falsiness of the value. What should be truthy/falsey is pretty contentious, but for the purposes of this explanation, an empty string is falsey and a non-empty string is truthy.

So, what is happening in the expression `"<" || "<a:"`, is it's evaluating the truthiness of `"<"` which is true, and ignoring the right side of `||` because of the aforementioned short circuit evaluation. Meaning `"<" || "<a:"` will simply evaluate to `"<"`. All that to say `if (!emoji.startsWith("<" || "<a:"))` is equivalent to `if (!emoji.startsWith("<"))` because `"<"` is always truthy, meaning `"<" || "<a:"` will evaluate to `"<"`.

Instead, it would need to be `emoji.startsWith("<") || emoji.startsWith("<a:")`, which is entirely irrelevant anyway because if `emoji` started with `<` it would have to start with `<a:` anyway.

Reply:
That makes so much sense, and I probably would have never caught this either. I just committed the change, and I actually moved it first so it doesn't even attempt converting it if the emoji isn't custom.